src/coreclr/vm/prestub.cpp

```diff
@@ -834,13 +835,24 @@ PCODE MethodDesc::JitCompileCodeLockedEventWrapper(PrepareCodeConfig* pConfig, J
             &methodSignature);
 #endif //FEATURE_EVENT_TRACE
 
-        pCode = JitCompileCodeLocked(pConfig, pilHeader, pEntry, &sizeOfCode);
+        pCode = JitCompileCodeLocked(pConfig, pilHeader, pEntry, &sizeOfCode, &isInterpreterCode);
```

Comment:
I have an idea - if we moved the creation of the interpreter precode from the `JitCompileCodeLocked` to this method after the call to the `ETW::MethodLog::MethodJitted`, then we would not need to do this.
This is the only caller of the `JitCompileCodeLocked`.

Reply:
Ah, that call to SetNativeCode is well hidden in there, I've missed it.